# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: servo/ports/geckolib/glue.rs
# Commit: 8a94c155e35c
# Full Hash: 8a94c155e35c69a524e0ecb17674bb5afa9c5ad3
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 16:30:15
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 1 - support rem, ch, ic and cap units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180618
# ==============================================================================

diff -r cd564dee524a -r 8a94c155e35c servo/ports/geckolib/glue.rs
--- a/servo/ports/geckolib/glue.rs	Mon Sep 11 23:42:37 2023 +0000
+++ b/servo/ports/geckolib/glue.rs	Mon Sep 11 23:50:11 2023 +0000
@@ -5352,6 +5352,18 @@
         structs::nsCSSUnit::eCSSUnit_XHeight => {
             NoCalcLength::FontRelative(FontRelativeLength::Ex(value))
         },
+        structs::nsCSSUnit::eCSSUnit_RootEM => {
+            NoCalcLength::FontRelative(FontRelativeLength::Rem(value))
+        },
+        structs::nsCSSUnit::eCSSUnit_Char => {
+            NoCalcLength::FontRelative(FontRelativeLength::Ch(value))
+        },
+        structs::nsCSSUnit::eCSSUnit_Ideographic => {
+            NoCalcLength::FontRelative(FontRelativeLength::Ic(value))
+        },
+        structs::nsCSSUnit::eCSSUnit_CapHeight => {
+            NoCalcLength::FontRelative(FontRelativeLength::Cap(value))
+        },
         structs::nsCSSUnit::eCSSUnit_Pixel => NoCalcLength::Absolute(AbsoluteLength::Px(value)),
         structs::nsCSSUnit::eCSSUnit_Inch => NoCalcLength::Absolute(AbsoluteLength::In(value)),
         structs::nsCSSUnit::eCSSUnit_Centimeter => {